Changes around context expiration naming and types (#2033)
Azure/azure-sdk-for-cpp: sdk/core/azure-core/CHANGELOG.md

  - `SetHeader(uint8_t const* const first, uint8_t const* const last)`.
  - `GetMajorVersion`.
  - `GetMinorVersion`.
- Changes to `Azure::Core::Context`:
  - Renamed `Get()` to `GetValue()`.
  - Renamed `WithValue()` to `CreateWithValue()`.

>   - Renamed `WithDeadline()` to `CreateWithExpiration()`, changed input parameter type to `Azure::DateTime`.

Reviewer: I feel the old names are better than the new ones.

Because for `Create`, I have the feeling that something is going to be created out of thin air. Like `CreateBlob`, there was nothing, and now we have a blob.

Reviewer 2: I like GetValue although I'm hoping for a GetValue(key, T valueIfKeyNotFound) function and then the removal of HasValue. Or, I'll take a bool TryGetValue(key, T* valueIfFound)

(Create)WithValue actually does create a Context on the heap "out of thin air". I prefer the old name too but with documentation that explains that a heap allocation occurs.

I also prefer no WithDeadline without Create and without Expiration. Accepting a DateTime is OK with me.

Reviewer 2: I looked-up deadline & expiration in the dictionary. I think deadline is better because of this: https://www.powerthesaurus.org/deadline/definitions

The Context holds "the point in time at which something must be completed".

>   - Renamed `WithDeadline()` to `CreateChildContext()`, and changed input parameter type to `Azure::DateTime`.

Reviewer 3: Revert the rename.

>   - Renamed `WithValue()` to `CreateChildContext()`.
>   - Renamed `WithDeadline()` to `CreateChildContext()`, and changed input parameter type to `Azure::DateTime`.

Author: ```suggestion
  - Changed input parameter type of `WithDeadline()` to `Azure::DateTime`.
```

- Added `Azure::Core::Operation<T>::GetRawResponse()`.
- Added `Azure::Core::PackageVersion`.
- Added support for logging to console when `AZURE_LOG_LEVEL` environment variable is set.

> - Added `Azure::Core::Context::GetExpiration()`.

Reviewer 2: GetDeadline

- Added `Azure::Core::Operation<T>::GetRawResponse()`.
- Added `Azure::Core::PackageVersion`.
- Added support for logging to console when `AZURE_LOG_LEVEL` environment variable is set.

> - Added `Azure::Core::Context::GetDeadline()`.

Reviewer 3: This is in the wrong release. Fixed in https://github.com/Azure/azure-sdk-for-cpp/pull/2069
